### PR TITLE
feat(Wizard): improve accessibility and forward ref

### LINKED ISSUE #277: Wizard: improve a11y and keyboard navigation
Wizard should have keyboard navigation similar to HorizontalTabs, and some aria attributes.

### PR DESCRIPTION
**Pull request by mayank99**
- Added tabIndex and onKeyDown to clickable Steps.
- Changed aria-current=true to aria-current=step.
- Forwarded ref to allow user to modify aria attributes on any level.
- Updated itwinui-css to 0.27.3.

Closes #277

### CODE REVIEW

File: src/core/Wizard/Wizard.tsx [hunk 1]
```diff
@@ -55,43 +55,43 @@ const defaultWizardLocalization: WizardLocalization = {
 };
 
 /**
- * A Wizard.
- * Give it some steps.
- * Tell it what step to display.
- * Set the type if you'd like.
- * That's about it.
+ * A wizard displays progress through a sequence of logical and numbered steps.
```
Comment (veekeys) [inline src/core/Wizard/Wizard.tsx:58]: Kinda liked that previous description.. Was fun to read..But maybe this is more professional.

File: src/core/Wizard/Step.tsx [hunk 2]
```diff
@@ -77,7 +77,8 @@ export const Step = (props: StepProps) => {
         ...style,
       }}
       onClick={onCompletedClick}
-      aria-current={isActive}
+      aria-current={isActive ? 'step' : undefined}
+      tabIndex={0}
```
Comment (veekeys) [inline src/core/Wizard/Step.tsx:81]: I do not think all he steps should be focusable.. Only the interactive ones, IMO (previous).
Now styling is awful for current and next ones:
image
image
Workflow should not have it at all.

image

Comment (veekeys) [reply src/core/Wizard/Step.tsx:81]: Also, keyboard click does not work :(

Comment (veekeys) [reply src/core/Wizard/Step.tsx:81]: Test needed

Comment (mayank99) [reply src/core/Wizard/Step.tsx]: Fixed all

**Review by veekeys — APPROVED** [review]

**Review by saaaaaally — APPROVED** [review]

**Pull request merged**